ledger/src/blockstore_db.rs

```diff
@@ -1995,13 +1998,26 @@ pub struct LedgerColumnOptions {
     // Determine the way to compress column families which are eligible for
     // compression.
     pub compression_type: BlockstoreCompressionType,
+
+    // Control the number of performance samples in a million for RocksDB reads
+    // and writes.
+    pub perf_context_samples_in_1m: u32,
+
+    // A counter to determine whether to sample the current RocksDB read operation.
+    pub perf_read_counter: Arc<AtomicU32>,
+
+    // A counter to determine whether to sample the current RocksDB write operation.
+    pub perf_write_counter: Arc<AtomicU32>,
```

Comment:
Probably `AtomicUsize` would be better.
Also I don't think any of these fields need to be `pub`.

Reply:
These don't need to be `pub`, but rust will complain about accessing private fields when using its `::default()`.  Looks like I will need to use a builder pattern in order to make them non-pub.  If that's the case I will create a separate PR.

Reply:
Is `pub(self)` enough?

Reply:
Doesn't look like `pub(self)` work, but thanks for the suggestion, I didn't know about such syntax!  I will try to find out whether there is any scope restriction that works.

```
error[E0451]: field `perf_read_counter` of struct `blockstore_db::LedgerColumnOptions` is private
    --> ledger/src/blockstore.rs:4351:23
     |
4351 |                     ..LedgerColumnOptions::default()
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ field `perf_read_counter` is private

error[E0451]: field `perf_write_counter` of struct `blockstore_db::LedgerColumnOptions` is private
    --> ledger/src/blockstore.rs:4351:23
     |
4351 |                     ..LedgerColumnOptions::default()
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ field `perf_write_counter` is private
```

Reply:
Fixed in #24763 by implementing `new()`.